source/wifi/wifi_hal.c

```diff
@@ -9490,7 +9490,36 @@ INT wifi_getHalCapability(wifi_hal_capability_t *cap)
             printf("%s: getRadioCapabilities idx = %d\n", __FUNCTION__, radioIndex);
             return RETURN_ERR;
         }
+
+        for (j = 0; j < cap->wifi_prop.radiocap[radioIndex].maxNumberVAPs; j++)
+        {
+            if (iter >= MAX_NUM_RADIOS * MAX_NUM_VAP_PER_RADIO)
+            {
+                 printf("%s: to many vaps for index map (%d)\n", __func__, iter);
+                 return RETURN_ERR;
+            }
+            iface_info = &cap->wifi_prop.interface_map[iter];
+            iface_info->phy_index = radioIndex; // XXX: parse phyX index instead
+            iface_info->rdk_radio_index = radioIndex;
+            memset(output, 0, sizeof(output));
+            if (wifi_getRadioIfName(radioIndex, output) == RETURN_OK)
+            {
+                strncpy(iface_info->interface_name, output, sizeof(iface_info->interface_name) - 1);
+            }
+            // TODO: bridge name
+            // TODO: vlan id
+            // TODO: primary
+            iface_info->index = radioIndex;
+            iface_info->index = array_index_to_vap_index(radioIndex, j);
+            memset(output, 0, sizeof(output));
```

Comment:
Probably not needed

Reply:
as discussed offline: this is only because of limited trust to the next function call.